src/Symfony/Component/Runtime/Tests/phpt/application.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+$_SERVER['SCRIPT_FILENAME'] = __FILE__;
+
+require __DIR__.'/../../vendor/symfony/runtime/autoload.php';
+
+return function (array $context) {
+    $command = new Command('go');
+    $command->setCode(function (InputInterface $input, OutputInterface $output) use ($context) {
+        $output->write('OK Application '.$context['SOME_VAR']);
```

Comment:
What will be the advantage of using `$context` rather than `$_SERVER` directly here?

Reply:
To separate infrastructure. The same reasons to use `Request` instead of `$_GET` or `$_POST`. Testability and other good things

Reply:
> What will be the advantage of using $context rather than $_SERVER directly here?

decoupling from global state, the very purpose of this component :)
this means the context could be computed from something else, e.g. a long running loop.